geospaas/base_viewer/views.py

```diff
@@ -24,30 +23,18 @@ def instantiation_and_set_form_defaults(self):
         self.forms = [None] * len(self.form_class)
         # loop for instantiating with defaults
         for counter, element_form_class in enumerate(self.form_class):
-            # similar default values can be set via *elif* at the bottom of this statement
-            if 'time_coverage_start' in element_form_class.Meta.fields:
-                self.forms[counter] = element_form_class(
-                    {'time_coverage_start':
-                     timezone.datetime(2000, 1, 1, tzinfo=timezone.utc).date(),
-                     'time_coverage_end': timezone.now().date(),
-                     })
-            # elif: ######## code here for other default values
-                    ######## code here for other default values
-            else:  # instantiating without default values
-                self.forms[counter] = element_form_class({})
-
+            self.forms[counter] = element_form_class({})  # instantiation
+            self.forms[counter].set_defaults()  # setting the defaults
```

Comment:
If SearchForm.__init__() calls set_default(). This is not needed.
```suggestion
```

Reply:
according to refusal of above suggestion, this is also denied consequently.

Reply:
It needed some converting of data of POST which makes code more complicated than the current one

Reply:
The call to this method can be avoided if you use a standard `Form` with dynamic initial values, as suggested in an other comment